Tasks/KubernetesManifestV0/src/CanaryDeploymentHelper.ts

```diff
@@ -0,0 +1,105 @@
+const CANARY_LABEL_VALUE = "canary";
+const CANARY_VERSION_LABEL = "azure-pipelines/version";
+
+export function calculateReplicaCountForCanary(inputObject: any, percentage: number) {
+    var inputReplicaCount = helper.getReplicaCount(inputObject);
+    return Math.round((inputReplicaCount * percentage) / 100);
+}
+
+export function isDeploymentEntity(kind: string): boolean {
+    if (!kind) {
+        throw new Error("Kind is not defined");
+    }
+
+    var temp = kind.toUpperCase();
+    return temp === helper.KubernetesWorkload.Pod.toUpperCase() ||
+        temp === helper.KubernetesWorkload.Replicaset.toUpperCase() ||
+        temp === helper.KubernetesWorkload.Deployment.toUpperCase() ||
+        temp === helper.KubernetesWorkload.StatefulSet.toUpperCase() ||
+        temp === helper.KubernetesWorkload.DaemonSet.toUpperCase();
+}
+
+export function getNewBaselineResource(stableObject: any, replicas: number): object {
+    return getNewCanaryObject(stableObject, replicas, BASELINE_LABEL_VALUE);
+}
+
+export function getNewCanaryResource(inputObject: any, replicas: number): object {
+    return getNewCanaryObject(inputObject, replicas, CANARY_LABEL_VALUE);
+}
+
+function getNewCanaryObject(inputObject: any, replicas: number, type: string): object {
+    var newObject = JSON.parse(JSON.stringify(inputObject));
+
+    // Updating name
+    newObject.metadata.name = type === CANARY_LABEL_VALUE ? getCanaryResourceName(inputObject.metadata.name) :
+        getBaselineResourceName(inputObject.metadata.name);
+
+    // Adding labels and annotations.
+    addCanaryLabelsAndAnnotations(newObject, type);
+
+    // Updating no. of replicas
+    if (newObject.kind.toUpperCase() != helper.KubernetesWorkload.Pod.toUpperCase() &&
+        newObject.kind.toUpperCase() != helper.KubernetesWorkload.DaemonSet.toUpperCase()) {
+        newObject.spec.replicas = replicas;
+    }
+
+    return newObject;
+}
+
+function addCanaryLabelsAndAnnotations(inputObject: any, type: string) {
+    var newLabels = new Map<string, string>();
+    newLabels[CANARY_VERSION_LABEL] = type;
+
+    helper.updateObjectLabels(inputObject, newLabels, false);
+    helper.updateObjectAnnotations(inputObject, newLabels, false);
+    helper.updateSelectorLabels(inputObject, newLabels, false);
+    helper.updateSpecLabels(inputObject, newLabels, false);
+}
+
+export function applyResource(kubectl: Kubectl, inputObjects: any[]) {
+    let newFilePaths = [];
+    inputObjects.forEach((inputObject: any) => {
+        var filePath = inputObject.kind + "_" + inputObject.metadata.name;
+        var inputObjectString = JSON.stringify(inputObject);
+        const tempDirectory = utils.getTempDirectory();
+        let fileName = path.join(tempDirectory, path.basename(filePath));
+        fs.writeFileSync(
+            path.join(fileName),
+            inputObjectString);
+        newFilePaths.push(fileName);
+    });
+
+    var result = null;
+    if (newFilePaths.length > 0) {
```

Comment:
Would this ever be the case? If length `== 0` the task would have thrown an error earlier only right? 
Or move the check to `inputObjects.length > 0` instead?